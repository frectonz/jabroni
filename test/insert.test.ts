import { makeWebSocketFetch, nanoid, snapshotTest } from "./base.ts";

const $fetch = await makeWebSocketFetch({
  url: "ws://127.0.0.1:3030",
  connectionCount: 10,
});

Deno.test(
  "insert a row into a table",
  snapshotTest($fetch, {
    type: "InsertRow",
    table: "employees",
    data: {
      FirstName: "test",
      LastName: "tester",
      Address: "11120 Jasper Ave NW",
      BirthDate: "1962-02-18 00:00:00",
      City: "Edmonton",
      Country: "Canada",
      Email: "test@test.com",
      Fax: "+1 (780) 428-3457",
      HireDate: "2002-08-14 00:00:00",
      Phone: "+1 (780) 428-9482",
      PostalCode: "T5K 2N1",
      ReportsTo: null,
      State: "AB",
      Title: "General Manager",
    },
    request_id: nanoid(),
  }),
);

Deno.test(
  "update a row",
  snapshotTest($fetch, {
    type: "UpdateRow",
    table: "employees",
    key: 9,
    data: {
      City: "Addis",
      Country: "Ethiopia",
    },
    request_id: nanoid(),
  }),
);

Deno.test(
  "get single row",
  snapshotTest($fetch, {
    type: "GetRow",
    table: "employees",
    key: 9,
    select: ["city", "country"],
    request_id: nanoid(),
  }),
);

Deno.test(
  "delete a row from a table",
  snapshotTest($fetch, {
    type: "DeleteRow",
    table: "employees",
    key: 9,
    request_id: nanoid(),
  }),
);
