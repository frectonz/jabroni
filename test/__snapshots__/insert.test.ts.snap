export const snapshot = {};

snapshot[`insert a row into a table 1`] = `
{
  data: {
    inserted_rows: 1,
    request_id: "",
    table: "employees",
    type: "InsertRow",
  },
}
`;

snapshot[`delete a row from a table 1`] = `
{
  data: {
    deleted_rows: 1,
    request_id: "",
    table: "employees",
    type: "DeleteRow",
  },
}
`;
